src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -174,24 +163,23 @@ public void subscribeToCertificateUpdates(GetCertificateRequest getCertificateRe
                     getCertificateRequest.getCertificateRequestOptions().getCertificateType();
             // TODO: Should be configurable
             KeyPair keyPair = CertificateStore.newRSAKeyPair(4096);
-            X509Certificate[] caCertificates = getX509CACertificates();
```

Comment:
The issue is that we generate the consumer once, when an external component subscribe and then if the `caCertificates` change, we don't return the updated values because we are effectively creating a closure.